backend/asmgen.ml

```diff
@@ -298,15 +299,22 @@ let compile_fundecl ~ppf_dump ~funcnames fd_cmm =
             Cfg_regalloc_validate.Description.create (Cfg_with_liveness.cfg_with_layout cfg)
         in
         cfg
-        ++ Profile.record ~accumulate:true "cfg_irc" Cfg_irc.run
+        ++ begin match regalloc with
+          | IRC -> Profile.record ~accumulate:true "cfg_irc" Cfg_irc.run
+          | LS -> Profile.record ~accumulate:true "cfg_ls" Cfg_ls.run
+          | Upstream -> assert false
+        end
         ++ Cfg_with_liveness.cfg_with_layout
-        ++ Profile.record ~accumulate:true "cfg_validate_description" (Cfg_regalloc_validate.run cfg_description)
+        ++ begin match cfg_description with
+          | None -> Fun.id
+          | Some cfg_description -> Profile.record ~accumulate:true "cfg_validate_description" (Cfg_regalloc_validate.run cfg_description)
+        end
```

Comment:
It would be nice to simplify this slightly more, `Cfg_regalloc_validate.run` can take a `Cfg_description.t option`.